src/main/java/org/jetbrains/bsp/bazel/server/BepServer.java

```diff
@@ -165,62 +158,43 @@ private void processFinishedEvent(BuildEventStreamProtos.BuildFinished buildFini
     bspClient.onBuildTaskFinish(finishParams);
   }
 
-  private void processCompletedEventIfPresent(BuildEventStreamProtos.BuildEvent event) {
+  private void processCompletedEvent(BuildEventStreamProtos.BuildEvent event) {
     if (event.hasCompleted()) {
-      processCompletedEvent(event.getCompleted());
+      consumeCompletedEvent(event.getCompleted());
     }
   }
 
-  private void processCompletedEvent(BuildEventStreamProtos.TargetComplete targetComplete) {
+  private void consumeCompletedEvent(BuildEventStreamProtos.TargetComplete targetComplete) {
     List<OutputGroup> outputGroups = targetComplete.getOutputGroupList();
-
     if (outputGroups.size() == 1) {
-      BuildEventStreamProtos.OutputGroup outputGroup = outputGroups.get(0);
-
-      if (outputGroup.getName().equals("scala_compiler_classpath_files")) {
-        for (BuildEventStreamProtos.BuildEventId.NamedSetOfFilesId fileSetId :
-            outputGroup.getFileSetsList()) {
-          for (BuildEventStreamProtos.File file :
-              namedSetsOfFiles.get(fileSetId.getId()).getFilesList()) {
-            URI protoPathUri;
-            try {
-              protoPathUri = new URI(file.getUri());
-            } catch (URISyntaxException e) {
-              throw new RuntimeException(e);
-            }
-
-            // TODO(gerardd) the body of try-catch will be extracted
-            try {
-              List<String> lines =
-                  com.google.common.io.Files.readLines(
-                      new File(protoPathUri), StandardCharsets.UTF_8);
-
-              for (String line : lines) {
-                List<String> parts = Splitter.on("\"").splitToList(line);
-                if (parts.size() != 3) {
-                  throw new RuntimeException("Wrong parts in sketchy textproto parsing: " + parts);
-                }
+      OutputGroup outputGroup = outputGroups.get(0);
+      if (outputGroup.getName().equals(Constants.SCALA_COMPILER_CLASSPATH_FILES)) {
+        processFileSets(outputGroup);
+      }
+    }
+  }
 
+  // TODO FOR REVIEW: better naming suggestions?
+  private void processFileSets(OutputGroup outputGroup) {
```

Comment:
This is used to fetch the scala compiler classpath. Perhaps calling it `fetchScalaJars` , `getScalaJars`, `getScalaClasspath` sound more interesting.